File: yandex_backend/26. Набор символов/solution.cpp
#include <iostream>
#include <string>
#include <set>
#include <climits>

using namespace std;

int main() {
  string s;
  getline(cin, s);

  string c_str;
  getline(cin, c_str);
  set<char> c(c_str.begin(), c_str.end());

  int n = s.length();
  int min_len = INT_MAX;

  for (int i = 0; i < n; ++i) {
    for (int j = i; j < n; ++j) {
      string sub = s.substr(i, j - i + 1);
      set<char> sub_set(sub.begin(), sub.end());
      if (sub_set == c) {
        min_len = min(min_len, (int)sub.length());
      }
    }
  }

  if (min_len == INT_MAX) {
    cout << 0 << endl;
  } else {
    cout << min_len << endl;
  }

  return 0;
}

File: yandex_backend/26. Набор символов/solution.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    public static void Main(string[] args)
    {
        string s = Console.ReadLine();
        string c_str = Console.ReadLine();
        HashSet<char> c = new HashSet<char>(c_str.ToCharArray());
        int n = s.Length;
        int minLen = int.MaxValue;

        for (int i = 0; i < n; i++)
        {
            for (int j = i; j < n; j++)
            {
                string sub = s.Substring(i, j - i + 1);
                HashSet<char> subSet = new HashSet<char>(sub.ToCharArray());
                if (subSet.SetEquals(c))
                {
                    minLen = Math.Min(minLen, sub.Length);
                }
            }
        }

        if (minLen == int.MaxValue)
        {
            Console.WriteLine(0);
        }
        else
        {
            Console.WriteLine(minLen);
        }
    }
}